compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -141,7 +141,8 @@ public SemanticModel getSemanticModel(ModuleId moduleId) {
         // If the states do not match, then this is a illegal state exception.
         if (moduleContext.compilationState() != ModuleCompilationState.COMPILED) {
             throw new IllegalStateException("Semantic model cannot be retrieved when the module is in " +
-                    "compilation state '" + moduleContext.compilationState().name() + "'. " +
+                    "compilation state'" + (moduleContext.compilationState() == null ? ""
+                    : (" " + moduleContext.compilationState().name())) + "'. " +
```

Comment:
If `moduleContext.compilationState() == null` shall we just say

```
Semantic model cannot be retrieved.
```

Reply:
But shouldn't we say that compilation is pending even though the state is not mentioned?

Reply:
If it is null doesn't it just say 

```
Semantic model cannot be retrieved when the module is in''. 
```

with the current change? :thinking:

Reply:
It would say,
> Semantic model cannot be retrieved when the module is in compilation state. This is an internal error which will be fixed in a later release.

Just updated to remove `

Reply:
<img width="1142" alt="Screenshot 2022-08-17 at 12 20 22" src="https://user-images.githubusercontent.com/15815199/185053590-a9e24a24-26ed-4883-8a46-b79f0bd6dbb6.png">

Reply:
Isn't it incomplete though? The state which is usually specified within the quotes isn't specified here?

Reply:
When we are doing a cyclic import, the state won't be set, but a resolution diagnostic will be added. Hence, its actually its compilation is pending. Here what I meant by `compilation state` is `compilation pending`. When the compilation state is set, the enum value will be contacted here. Wouldn't it make sense though?

Reply:
Yeah, but then shouldn't we say 

```
Semantic model cannot be retrieved when the module is in compilation state 'PENDING'.
```

Right now, it is like an incomplete sentence, right?